Drive/rover_drive_system.hpp

```diff
@@ -150,19 +150,191 @@ class RoverDriveSystem
   /// @param speed the new movement speed of the rover
   void SetWheelSpeed(units::angular_velocity::revolutions_per_minute_t speed)
   {
-    // TODO: Implement linear interpolation (exponential moving average)
+    //Want to make sure Uptime has atleast 100 ms and don't want to waste 10ms on the first iteration
+      auto next_same_time = (Uptime() + 100ms) % 10;
+      next_same_time = Uptime() - next_same_time;
+
+
+      long double lerpSpeed_leftWheel;
+      long double lerpSpeed_rightWheel;
+      long double lerpSpeed_backWheel;
+
+      long double currentSpeed_leftWheel = static_cast<long double>(left_wheel_.GetSpeed());
+      long double currentSpeed_rightWheel = static_cast<long double>(right_wheel_.GetSpeed()); //GetSpeed);
+      long double currentSpeed_backWheel = static_cast<long double>(back_wheel_.GetSpeed());
+
+      long double new_speed = static_cast<long double>(speed);
+
+      //should be okay for almost all cases with 9 loops
+      for(int i = 0; i < 9; i++)
+      {
+        if((next_same_time % 10ms) == 0ms)
```

Comment:
I think this logic might be somewhat problematic since %10 == 0ms is highly unlikely. What if your timer is always 1ms to early or 1ms too late you'll never call this if statement. A possible solution might be to only lerp the wheels if at least 10ms has passed since the last time the wheels lerped?